### PR TITLE
feat(nimbus): format JSON for better readability in branch configuration

### LINKED ISSUE #11722: format branch config json
Branch configs are long and not readable, format json to view it better

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4950)

### PR DESCRIPTION
**Pull request by RJAK11**
Because

* Branch configuration values were displayed in paragraph format, making them difficult to read

This commit

* Pretty-prints JSON with indentation for improved readability in the branch configuration

Fixes #11722

image
image

### CODE REVIEW

File: experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py [hunk 1]
```diff
@@ -65,3 +67,21 @@ def format_not_set(value):
...
+        return format_not_set(value)
+
+    try:
+        parsed_json = json.dumps(json.loads(value), indent=2)
+    except (json.JSONDecodeError, TypeError):
+        parsed_json = value
+
+    return mark_safe(
+        '<pre class="text-monospace" style="white-space: pre-wrap; '
+        'word-wrap: break-word;">' + parsed_json + "</pre>"
```
Comment (jaredlockhart) [inline experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py:86]: Oh right here, don't use concatenation to join strings 🙏 , use [fstrings](https://fstring.help/).

File: experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py [hunk 2]
```diff
@@ -65,3 +67,21 @@ def format_not_set(value):
     if value in [None, "", "NOT SET"]:
         return mark_safe('<span class="text-danger">Not set</span>')
     return value
+
+
+@register.filter(name="format_json")
+def format_json(value):
+    """Formats the JSON value for display by pretty-printing the JSON,
+    and if the JSON is empty or None, it calls format_not_set."""
+    if value in [None, "", "NOT SET"]:
```
Comment (jaredlockhart) [inline experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py:76]: If we're gonna do this check twice

```py
if value in [None, "", "NOT SET"]
```

Then we should move that to a constant the top of the file like

```py
NOT_SET_VALUES = [None, "", "NOT SET"]

...

if value not in NOT_SET_VALUES:
```

**Comment by yashikakhurana** [comment]
was about to take a look, but Jared did it :D +1 to his points

File: experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py (hunk 2 above)
Comment (yashikakhurana) [inline experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py:76]: Also it just occurred to me that you can avoid this and apply multiple filters `  <code>{{ feature_value.value|format_not_set|format_json }}</code>`

**Review by yashikakhurana — COMMENTED** [review]
Overall looks good, just one minor suggestion

File: experimenter/experimenter/nimbus_ui_new/templates/nimbus_experiments/detail.html [hunk 3]
```diff
@@ -275,7 +275,9 @@ <h6>
                 {% for feature_value in branch.feature_values.all %}
                   <tr>
                     <th>{{ feature_value.feature_config.name|format_not_set }} Value</th>
-                    <td colspan="3">{{ feature_value.value|format_not_set }}</td>
+                    <td colspan="3">
+                      <code>{{ feature_value.value|format_not_set|format_json }}</code>
```
Comment (jaredlockhart) [inline experimenter/experimenter/nimbus_ui_new/templates/nimbus_experiments/detail.html:279]: Oh that's an even better design 🎉 🎉 🎉 🎉 🎉

**Review by yashikakhurana — APPROVED** [review]
@RJAK11 thank you, tested locally, it looks great 🎉 🎉 🎉

**Pull request merged**